Bug: GCP Storage API fails to compile due to visibility and borrow checker errors

The `gcp_storage.rs` module fails to compile due to missing visibility modifiers and Rust borrow checker violations.

---

## Compilation Errors

### Visibility Errors (E0425, E0433)

error[E0425]: cannot find type `GoogleStorage` in this scope  
 --> src/tests/gcp_storage_operations.rs

error[E0433]: failed to resolve: use of undeclared type `GoogleStorage`  
 --> src/tests/gcp_storage_operations.rs


### Borrow Checker Errors (E0716)

error[E0716]: temporary value dropped while borrowed  
 --> src/gcp/gcp_apis/storage/gcp_storage.rs

error[E0716]: temporary value dropped while borrowed  
 --> src/gcp/gcp_apis/storage/gcp_storage.rs

---

## Root Cause

- `GoogleStorage` struct and its associated methods are not public, making them inaccessible from test modules.
- Temporary `Value::String` values are created inline and dropped while still borrowed.

---

## Affected File

rustcloud/src/gcp/gcp_apis/storage/gcp_storage.rs

---

## Expected Behavior

The project should compile successfully without errors.

---

## Suggested Fix

### 1. Add visibility modifiers

pub struct GoogleStorage  
pub fn new()  
pub async fn create_disk()

### 2. Fix borrow checker issue

Store temporary values before inserting into the map:

let location = Value::String("us-central1".to_string());  
map.insert("location".to_string(), location);

---

@PratikDhanave

--- a/rustcloud/src/gcp/gcp_apis/storage/gcp_storage.rs
+++ b/rustcloud/src/gcp/gcp_apis/storage/gcp_storage.rs
@@ -3,20 +3,20 @@ use reqwest::header::AUTHORIZATION;
 use serde_json::Value;
 use std::collections::HashMap;
 
-struct GoogleStorage {
+pub struct GoogleStorage {
     client: reqwest::Client,
     base_url: String,
 }
 
 impl GoogleStorage {
-    fn new() -> Self {
+    pub fn new() -> Self {
         GoogleStorage {
             client: reqwest::Client::new(),
             base_url: "https://www.googleapis.com/compute/v1".to_string(),
         }
     }
 
-    async fn create_disk(
+    pub async fn create_disk(
         &self,
         request: HashMap<String, Value>,
     ) -> Result<HashMap<String, Value>, reqwest::Error> {
@@ -97,17 +97,13 @@ impl GoogleStorage {
             }
         }
 
-        option.insert(
-            "Zone",
-            &Value::String(format!("projects/{}/zones/{}", project_id, zone)),
-        );
-        option.insert(
-            "Type",
-            &Value::String(format!(
-                "projects/{}/zones/{}/diskTypes/{}",
-                project_id, zone, disk_type
-            )),
-        );
+        let zone_value = Value::String(format!("projects/{}/zones/{}", project_id, zone));
+        let type_value = Value::String(format!(
+            "projects/{}/zones/{}/diskTypes/{}",
+            project_id, zone, disk_type
+        ));
+        option.insert("Zone", &zone_value);
+        option.insert("Type", &type_value);
 
         let create_disk_json = serde_json::to_string(&option).unwrap();
         let url = format!(
@@ -136,7 +132,7 @@ impl GoogleStorage {
         Ok(response)
     }
 
-    async fn delete_disk(
+    pub async fn delete_disk(
         &self,
         request: HashMap<String, String>,
     ) -> Result<HashMap<String, Value>, reqwest::Error> {
@@ -165,7 +161,7 @@ impl GoogleStorage {
         Ok(response)
     }
 
-    async fn create_snapshot(
+    pub async fn create_snapshot(
         &self,
         request: HashMap<String, Value>,
     ) -> Result<HashMap<String, Value>, reqwest::Error> {

